Samples/WinMLSamplesGallery/WinMLSamplesGalleryNative/DXResourceBinding.cpp

```diff
@@ -0,0 +1,59 @@
+#define NOMINMAX
+#define WIN32_LEAN_AND_MEAN
+
+#include "pch.h"
+#include "DXResourceBinding.h"
+#include "DXResourceBinding.g.cpp"
+#include "stdafx.h"
+#include "ORTHelpers.h"
+
+using namespace winrt::Microsoft::AI::MachineLearning;
+
+#undef min
+
+Microsoft::WRL::ComPtr<ID3D12Resource> d3dResource;
+static std::optional<Ort::Session> preprocesingSession;
+static std::optional<Ort::Session> inferenceSession;
+D3D12Quad sample(800, 600, L"D3D12 Quad");
+
+namespace winrt::WinMLSamplesGalleryNative::implementation
+{
+    // Create ORT Sessions and launch D3D window in a separate thread
+	void DXResourceBinding::LaunchWindow() {
+        // Create ORT Sessions that will be used for preprocessing and classification
+        preprocesingSession = CreateSession(Win32Application::GetAssetPath(L"dx_preprocessor_efficient_net.onnx").c_str());
+        inferenceSession = CreateSession(Win32Application::GetAssetPath(L"efficientnet-lite4-11.onnx").c_str());
+
+        // Spawn the window in a separate thread
+        std::thread d3d_th(Win32Application::Run, &sample, 10);
```

Comment:
>std::thread [](http://example.com/codeflow?start=8&length=11)

`std::jthread` is generally more robust, because `std::thread` has this braindead default of calling std::terminate upon exiting scope if you don't join or detach (unfortunately it was too late to change, and even the C++ chairman Herb Sutter proposed fixing it)

[https://www.modernescpp.com/index.php/an-improved-thread-with-c-20](https://www.modernescpp.com/index.php/an-improved-thread-with-c-20)

In your case, it doesn't really matter because you're calling `detach` anyway, but for general copy&paste goodness.

Reply:
Done.